app/routines/add_email_to_user.rb

```diff
@@ -27,7 +27,12 @@ def exec(email_address_text, user, options={})
     email_address.save
     transfer_errors_from(email_address, { scope: :email_address }, true)
 
-    # The confirmation info won't be sent if already verified
-    run(SendContactInfoConfirmation, email_address)
+    # Unclaimed accounts only receive a welcome email
+    if user.is_unclaimed?
+      UnclaimedUserMailer.welcome(email_address).deliver
```

Comment:
This is great and thanks for adding this.  Now the trick is that these user's email addresses will never be confirmed!  Can you think it through and figure out the best way to confirm the email address?

Haven't thought this all the way through, but maybe the best way is just use the contact info confirmation mailer but change the text for the unclaimed case (to have the invitation language you wrote but next to a link that confirms their email address and then drops them off in account finalization or something).  Thoughts?